### ISSUE TITLE
t/live-https.t fails (Server closed connection)

### ISSUE THREAD

**Issue by eserte** [libwww-perl/Net-HTTP#57]
Right now t/live-https.t fails (various OS, various perl versions):
```
perl5.30.0-RC1 -Mblib t/live-https.t
1..6
# ----------------------------
# 200 OK
# Accept-Ranges: bytes
# Age: 369
# Cache-Control: public, max-age=900, stale-while-revalidate=90, stale-if-error=172800
# Connection: close
# Content-Length: 8396
# Content-Type: text/html
# Date: Sun, 12 May 2019 17:57:10 GMT
# ETag: "20cc-588b466c17ac0"
# Last-Modified: Sun, 12 May 2019 17:48:03 GMT
# Server: Apache/2.4.29 (Unix)
# Strict-Transport-Security: max-age=15724800;
# Vary: Accept-Encoding
# Via: 1.1 varnish
# X-Cache: HIT, HIT
# X-Cache-Hits: 1, 1
# X-Served-By: cache-hhn1539-HHN, cache-hel6827-HEL
# X-Timer: S1557683831.707778,VS0,VE1
#
ok 1 - success
ok 2
ok 3
Server closed connection without sending any data back at /home/cpansand/.cpan/build/2019051219/Net-HTTP-6.18-0/blib/lib/Net/HTTP/Methods.pm line 391.
# Looks like your test exited with 255 just after 3.
```

An strace log is probably not that useful with encrypted connections, but here it is:
```
18878 19:48:57.507596 write(5, "ok 1 - success\n", 15) = 15 <0.000022>
18878 19:48:57.507903 write(5, "ok 2\n", 5) = 5 <0.000019>
18878 19:48:57.508199 write(5, "ok 3\n", 5) = 5 <0.000019>
18878 19:48:57.508341 fcntl(3, F_GETFL) = 0x2 (flags O_RDWR) <0.000011>
18878 19:48:57.508406 write(3, "\27\3\3\0|\373\324\4\257\223\260:\326\326\24\272\317.FY\370\352+\363\331\243_\247|\273S\341\310\325\3\242\"*\0378\20\354\370\230\202\v\221\314\34\241\321A4\275\315\26\333\10\222\361\204\273B}K\0es\374\224:h\335O}$)\366\24\305\251\342Hy\271\361X\\\31\20\374\345\336\245L\325}Z\252\332\373\303\210+\363I\22\350\205\2548\254\375\254\\\254\255\316\37\256\f*\t\214\241\227\23\223\331", 129) = 129 <0.000049>
18878 19:48:57.508526 select(8, [3], NULL, NULL, {15, 0}) = 1 (in [3], left {14, 999998}) <0.000012>
18878 19:48:57.508585 read(3, "\25\3\3\0\32", 5) = 5 <0.000011>
18878 19:48:57.508621 read(3, "\234z0\263)\273\343B\215x\16r\16\17F\325\32P\240\360\201\240\261\316\272 ", 26) = 26 <0.000013>
18878 19:48:57.508674 write(2, "Server closed connection without sending any data back at /home/cpansand/.cpan/build/2019051219/Net-HTTP-6.18-0/blib/lib/Net/HTTP/Methods.pm line 391.\n", 151) = 151 <0.000018>
18878 19:48:57.509582 close(3)          = 0 <0.000091>
```

**Comment by eserte** [comment]
I think the problem is
```
# Connection: close
```
seen in the response above. A good response looks like this:
```
# ----------------------------
# 200 OK
# Accept-Ranges: bytes
# Age: 588
# Cache-Control: public, max-age=900, stale-while-revalidate=90, stale-if-error=172800
# Connection: keep-alive
# Content-Length: 8342
# Content-Type: text/html
# Date: Tue, 14 May 2019 20:23:59 GMT
# ETag: "2096-588dea56c2ac0"
# Last-Modified: Tue, 14 May 2019 20:12:03 GMT
# Server: Apache/2.4.29 (Unix)
# Strict-Transport-Security: max-age=15724800;
# Vary: Accept-Encoding
# Via: 1.1 varnish
# X-Cache: HIT, HIT
# X-Cache-Hits: 2, 3
# X-Served-By: cache-hhn1539-HHN, cache-fra19152-FRA
# X-Timer: S1557865440.876055,VS0,VE0
```
The test probably assumes that there's a working keep-alive connection, and fails if it isn't.

And why the different responses? There's a CDN or so involved here. On some of my smokers www.cpan.org resolves to 151.101.14.217 (where the test suite passes) and on some it's 151.101.246.217. To workaround the problem, just add
```
151.101.14.217 www.cpan.org
```
to `/etc/hosts`.

A solution could be to inspect the `Connection` value in the response, and skip the rest of the test script if it says `close`.

**Comment by oalders** [comment]
Fixed via #58
